Changes in the release section of the readme (#8259)
RasaHQ/rasa: README.md

1. **Go over the milestone and evaluate the status of any PR merging that's happening. Follow up with people on their
bugs and fixes.** If the release introduces new bugs or regressions that can't be fixed in time, we should discuss on

> Slack about this and take a decision on how to move forward. If the issue is not ready to be merged in time, we remove issue / PR from milestone and notify the PR owner and the product manager on Slack about it. The PR / issue owners are responsible for

Reviewer: ```suggestion
Slack about this and take a decision on how to move forward. If the issue is not ready to be merged in time, we remove the issue / PR from the milestone and notify the PR owner and the product manager on Slack about it. The PR / issue owners are responsible for
```

> communicating any issues which might be release relevant. Postponing the release should be considered as an edge case scenario based on the latest decision about [Time based releases](https://www.notion.so/rasa/Time-based-Releases-4861e429c73f402fa83dc06c0aba0435) and try to avoid it.

Reviewer: Mhm, I'd avoid to link a private Notion page in the Readme which is public. How about just keeping everything unti "based on"?

Reviewer 2: I agree with this, we shouldn't add private external links to the README of this repo

#### Release day! 🚀

1. **At the start of the day, post a small message on slack announcing release day!** Communicate you'll be handling
the release, and the time you're aiming to start releasing (again, no later than 4pm, as issues may arise and
cause delays). This message should be posted early in the morning and before moving forward with any of the steps of the release, 

>    in order to give enough time to people to check their PRs and Issues. That way they still have time in case they would like 

Reviewer: ```suggestion
   in order to give enough time to people to check their PRs and issues. That way they still have time in case they would like 
```

>    in order to give enough time to people to check their PRs and Issues. That way they still have time in case they would like 
>    to add something in the Milestone before the announced release time. Template of the slack message can be found [here](https://rasa-hq.slack.com/archives/C36SS4N8M/p1613032208137500?thread_ts=1612876410.068400&cid=C36SS4N8M).

Reviewer: The day of the release is in my opinion definitely too late to add things to the milestone. It's more that they can plan any remaining work or reach out if just tiny steps are missing for something to be merged (e.g. the CI run failed and they had to re-rerun)
```suggestion
   in order to give enough time to people to check their PRs and Issues. That way they can plan any remaining work. A template of the slack message can be found [here](https://rasa-hq.slack.com/archives/C36SS4N8M/p1613032208137500?thread_ts=1612876410.068400&cid=C36SS4N8M).
```

Reviewer: in my opinion we could also directly put the Slack message here.

Reviewer 2: I'd say we could keep the link to avoid making the instructions suuuuper long if not absolutely necessary. We have slack links elsewhere in the README anyway.

1. Make sure all dependencies are up to date (**especially Rasa SDK**)
    - For Rasa SDK that means first creating a [new Rasa SDK release](https://github.com/RasaHQ/rasa-sdk#steps-to-release-a-new-version) (make sure the version numbers between the new Rasa and Rasa SDK releases match)
    - Once the tag with the new Rasa SDK release is pushed and the package appears on [pypi](https://pypi.org/project/rasa-sdk/), the dependency in the rasa repository can be resolved (see below).
2. In case of a minor release, create a new branch that corresponds to the new release, e.g. 
   ```bash

>     git checkout -b 2.4.0

Reviewer: ```suggestion
    git checkout -b 2.4.x
```

    ```

> 3. Switch to the branch you want to cut the release from (`main` in case of a major, the branch you just created in case of a minor, the current feature branch for micro releases)

Reviewer: ```suggestion
3. Switch to the branch you want to cut the release from (`main` in case of a major, the `<major>.<minor>.x` branch for minors and micros)
```

    git checkout -b 1.2.x
    git push origin 1.2.x
    ```
7. After all the steps are completed and if everything goes well then we should see a message automatically posted in the company's Slack (`product` channel) like this [one](https://rasa-hq.slack.com/archives/C7B08Q5FX/p1614354499046600)
8. If no message appears in the channel then you can do the following checks:

>     - Check the workflows in [Github Actions](https://github.com/RasaHQ/rasa/actions) and make sure that the merged PR of the current release is completed successfully. To easily find your PR you can use the filters `event: push` and `branch: <version number>` (example on release 2.4 you can see [here](https://github.com/RasaHQ/rasa/actions/runs/643344876))

Reviewer: ```suggestion
    - Check the workflows in [Github Actions](https://github.com/RasaHQ/rasa/actions) and make sure that the merged PR of the current release is completed successfully. To find your PR you can use the filters `event: push` and `branch: <version number>` (example on release 2.4 you can see [here](https://github.com/RasaHQ/rasa/actions/runs/643344876))
```

    git checkout -b 1.2.x
    git push origin 1.2.x
    ```
7. After all the steps are completed and if everything goes well then we should see a message automatically posted in the company's Slack (`product` channel) like this [one](https://rasa-hq.slack.com/archives/C7B08Q5FX/p1614354499046600)
8. If no message appears in the channel then you can do the following checks:

>     - Check the workflows in [Github Actions](https://github.com/RasaHQ/rasa/actions) and make sure that the merged PR of the current release is completed successfully. To easily find your PR you can use the filters `event: push` and `branch: 2.x.x` (example on release 2.4 you can see [here](https://github.com/RasaHQ/rasa/actions/runs/643344876))

Reviewer: ```suggestion
    - Check the workflows in [Github Actions](https://github.com/RasaHQ/rasa/actions) and make sure that the merged PR of the current release is completed successfully. To easily find your PR you can use the filters `event: push` and `branch: <version number>` (example on release 2.4 you can see [here](https://github.com/RasaHQ/rasa/actions/runs/643344876))
```

    - If the workflow is not completed, then maybe try to re run the workflow in case that solves the problem

>     - If the problem persists then contact the infrastructure team

Reviewer 2: I'd say we should first check the logs and then contact infra 😉
